js_tests/admin/SelectFilter2.test.js

```diff
@@ -16,3 +16,131 @@ QUnit.test('init', function(assert) {
     assert.equal($('.selector-remove').text(), "Remove");
     assert.equal($('.selector-clearall').text(), "Remove all");
 });
+
+QUnit.test('filtering available options', function(assert) {
+    const $ = django.jQuery;
+    $('<form><select multiple id="select"></select></form>').appendTo('#qunit-fixture');
+    $('<option value="foo" title="Foo">Foo</option>').appendTo('#select');
+    $('<option value="bar" title="Bar">Bar</option>').appendTo('#select');
+    $('<option value="baz" title="Baz">Baz</option>').appendTo('#select');
```

Comment:
Try to use more realistic values in tests, maybe colors :thinking: 
```suggestion
    $('<option value="1" title="Red">Red</option>').appendTo('#select');
    $('<option value="2" title="Blue">Blue</option>').appendTo('#select');
    $('<option value="3" title="Green">Green</option>').appendTo('#select');
```